client/session/manager.go

```diff
@@ -0,0 +1,175 @@
+package session
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+	"sync"
+
+	"github.com/pkg/errors"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+)
+
+// Caller can invoke requests on the session
+type Caller interface {
+	Context() context.Context
+	Supports(method string) bool
```

Comment:
grpc wire definition only defines a request/response structure that is based on url(path). You can always check the service by checking part of that path. You could write `SupportsService` based on current info if you know the semantics of grpc URL, but I don't see a reason to add it atm when I don't see any use case for it. Avoiding helper would be more convenient but user code doesn't need to call it. I'd prefer to keep it in the terms of wire format to make sure that this is not abused with random strings.